databases/week3/Week3Homework.sql

```diff
@@ -0,0 +1,205 @@
+UPDATE meal 
+SET title="Ghee Dosai",
+description = "Ghee dosai with chutney"
+WHERE id=4;
+
+-- Delete a meal with any id, fx 1
+DELETE
+FROM meal
+WHERE id=5;
+
+-- queries for reservation table
+-- get all reservations
+SELECT *
+FROM reservation;
+
+-- Add a new reservation
+INSERT into reservation (id, number_of_guests, meal_id, created_date, contact_phonenumber, contact_name)
+values(4, 4, 3, "2021-04-25", 36665512, "subish");
+INSERT into reservation (id, number_of_guests, meal_id, created_date, contact_phonenumber, contact_name)
+values(5, 2, 2, "2021-04-25", 35476563, "Moni");
+
+-- Get a reservation with any id, fx 1
+SELECT *
+FROM reservation
+WHERE id=1;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation 
+SET contact_name = "krithikaa",
+number_of_guests=2
+WHERE id=2;
+
+-- Delete a reservation with any id, fx 1
+DELETE
+FROM reservation
+WHERE id=4;
+
+-- queries for review table
+-- Get all reviews
+SELECT *
+FROM review;
+
+-- Add a new review
+INSERT into review (id,title, description, meal_id, stars, created_date)
+values(4, "Dosai","chutney was not that good", 4, 2, "2021-04-28");
+INSERT into review (id,title, description, meal_id, stars, created_date)
+values(5, "veg biriyani","Too spicy", 1, 1, "2021-04-28");
+
+-- Get a review with any id, fx 1
+SELECT *
+FROM review
+WHERE id=2;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review 
+SET title = "Good biriyani",
+stars=5
+WHERE id=1;
+
+-- Delete a review with any id, fx 1
+DELETE
+FROM review
+WHERE id=4;
+
+-- functionality
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT *
+FROM meal
+WHERE price<100;
+
+-- Get meals that still has available reservations
+SELECT (meal.max_reservations - reservation.number_of_guests), meal.title
+FROM reservation
+INNER JOIN meal ON reservation.id=meal.id
+WHERE max_reservations - number_of_guests;
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT *
+FROM meal
+WHERE title LIKE "%biriyani%";
+
+-- Get meals that has been created between two dates
+SELECT meal.title, meal.description
+FROM meal
+WHERE created_date > "2021-04-23"
+AND created_date < "2021-04-29";
+
+-- Get only specific number of meals fx return only 5 meals
+SELECT meal.title, meal.description
+FROM meal
+LIMIT 2;
+
+-- Get the meals that have good reviews
+SELECT *
+FROM review
+WHERE stars > 3;
+
+-- Get reservations for a specific meal sorted by created_date
+SELECT meal.title, meal.description, reservation.created_date
+FROM meal
+JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.id = 1
+ORDER BY reservation.created_date ASC;
+
+-- Sort all meals by average number of stars in the reviews
+SELECT meal.id, meal.title, meal.description, avg(review.stars)
+FROM meal
+JOIN review ON meal.id = review.meal_id
+GROUP BY meal.id;
```

Comment:
`GROUP BY meal_id;`   might give you the desired output.

Reply:
Yes, implemented.